libobs/graphics/graphics.c

```diff
@@ -1273,7 +1273,7 @@ gs_swapchain_t *gs_swapchain_create(const struct gs_init_data *data)
 		return NULL;
 
 	if (new_data.num_backbuffers == 0)
-		new_data.num_backbuffers = 1;
+		new_data.num_backbuffers = 2;
```

Comment:
What purpose does this serve? Justify please.

Reply:
Can't use a flip model swap chain without at least two buffers. I verified that it fails with just one. The blt path still works with two instead of one, so I didn't want to complicate the code with extra conditional logic.

Reply:
Can DXGI_SWAP_CHAIN_FLAG_FRAME_LATENCY_WAITABLE_OBJECT be utilised here to offset the increased latency due to use of 2 backbuffers?

Reply:
Possibly, but writing code to block on a waiting primitive can sometimes be tricky to get right, and I feel like OBS has a higher risk of regression because there's various annoying factors involved: video output rate, monitor refresh rate, whether or not GPU encoding is involved, etc. The payoff just doesn't seem that great; maybe the preview updates a frame earlier if implemented correctly.

Also given the black'ish box nature of DXGI, It's not clear that latency is actually worsened by doing this. I'll try out various scenarios with my 240 FPS phone cam later out of curiosity.

Reply:
I moved the code around to make BufferCount 1 again for the legacy path, so I won't have to deal with behavior changes there.

Reply:
Recorded monitor/TV simultaneously and counted frames until monitor caught up. Both devices are 60 Hz. I suspect this is more or less a draw. I only checked 4 events on each side.

DISCARD, BufferCount=1 path (tested on Windows 10):
30/240 = 0.125 ms
30/240 = 0.125 ms
31/240 = 0.129 ms
31/240 = 0.129 ms

FLIP_DISCARD, BufferCount=2 (tested on Windows 10):
29/240 = 0.121 ms
29/240 = 0.121 ms
29/240 = 0.121 ms
29/240 = 0.121 ms